File: app/src/main/java/com/dertyp7214/appstore/components/CustomSnackbar.java
/*
 *
 *  * Copyright (c) 2018.
 *  * Created by Josua Lengwenath
 *
 */

/*
 * Copyright (c) 2018.
 * Created by Josua Lengwenath
 */

package com.dertyp7214.appstore.components;

import android.content.Context;
import android.graphics.Color;
import android.view.View;

import com.dertyp7214.appstore.ThemeStore;
import com.dertyp7214.appstore.Utils;
import com.google.android.material.snackbar.Snackbar;

import androidx.annotation.ColorInt;

public class CustomSnackbar {

    public static final int LENGTH_LONG = Snackbar.LENGTH_LONG;
    public static final int LENGTH_INDEFINITE = Snackbar.LENGTH_INDEFINITE;
    public static final int LENGTH_SHORT = Snackbar.LENGTH_SHORT;

    private Snackbar snackbar;
    private ThemeStore themeStore;
    private int color = 0;

    public CustomSnackbar(Context context){
        themeStore = ThemeStore.getInstance(context);
    }

    public CustomSnackbar(Context context, @ColorInt int color){
        themeStore = ThemeStore.getInstance(context);
        this.color = color;
    }

    public CustomSnackbar make(View view, CharSequence text, int duration){
        int backgroundColor = themeStore.getPrimaryColor();
        int textColor = themeStore.getPrimaryTextColor();
        if(color != 0)
            backgroundColor = color;
        if(Utils.isColorBright(backgroundColor))
            if(Utils.isColorBright(textColor))
                textColor = Color.BLACK;
        if(!Utils.isColorBright(backgroundColor))
            if(!Utils.isColorBright(textColor))
                textColor = Color.WHITE;
        snackbar = Snackbar.make(view, text, duration);
        View snackbarView = snackbar.getView();
        snackbarView.setBackgroundColor(backgroundColor);
        snackbar.setActionTextColor(textColor);
        /* TextView textView = snackbarView.findViewById(com.google.android.material.R.id.snackbar_text);
        textView.setTextColor(textColor); */
        return this;
    }

    public CustomSnackbar setAction(String text, View.OnClickListener listener){
        snackbar.setAction(text, listener);
        return this;
    }

    public CustomSnackbar setCallBack(Snackbar.Callback callBack){
        snackbar.addCallback(callBack);
        return this;
    }

    public void show(){
        snackbar.show();
    }

}

File: app/src/main/java/com/dertyp7214/appstore/components/CustomSnackbar.kt
/*
 *
 *  * Copyright (c) 2018.
 *  * Created by Josua Lengwenath
 *
 */

/*
 * Copyright (c) 2018.
 * Created by Josua Lengwenath
 */

package com.dertyp7214.appstore.components

import android.content.Context
import android.graphics.Color
import android.view.View
import androidx.annotation.ColorInt
import com.dertyp7214.appstore.ThemeStore
import com.dertyp7214.appstore.Utils
import com.google.android.material.snackbar.Snackbar

class CustomSnackbar {

    private var snackbar: Snackbar? = null
    private var themeStore: ThemeStore? = null
    private var color = 0

    constructor(context: Context) {
        themeStore = ThemeStore.getInstance(context)
    }

    constructor(context: Context, @ColorInt color: Int) {
        themeStore = ThemeStore.getInstance(context)
        this.color = color
    }

    fun make(view: View, text: CharSequence, duration: Int): CustomSnackbar {
        var backgroundColor = themeStore!!.primaryColor
        var textColor = themeStore!!.primaryTextColor
        if (color != 0)
            backgroundColor = color
        if (Utils.isColorBright(backgroundColor))
            if (Utils.isColorBright(textColor))
                textColor = Color.BLACK
        if (!Utils.isColorBright(backgroundColor))
            if (!Utils.isColorBright(textColor))
                textColor = Color.WHITE
        snackbar = Snackbar.make(view, text, duration)
        val snackbarView = snackbar!!.view
        snackbarView.setBackgroundColor(backgroundColor)
        snackbar!!.setActionTextColor(textColor)
        return this
    }

    fun setAction(text: String, listener: View.OnClickListener): CustomSnackbar {
        snackbar!!.setAction(text, listener)
        return this
    }

    fun setCallBack(callBack: Snackbar.Callback): CustomSnackbar {
        snackbar!!.addCallback(callBack)
        return this
    }

    fun show() {
        snackbar!!.show()
    }

    companion object {
        const val LENGTH_LONG = Snackbar.LENGTH_LONG
        const val LENGTH_INDEFINITE = Snackbar.LENGTH_INDEFINITE
        const val LENGTH_SHORT = Snackbar.LENGTH_SHORT
    }
}
